### PR TITLE
Decoupled neeto-package dependencies

### LINKED ISSUE #46: Decouple the neeto packages dependencies
We have open-sourced this package for the public, so it is not feasible to have a neeto dependency for this package.
When I installed this package at a non-neeto project 
Screenshot 2024-09-03 at 1 45 24 PM
it is instructing me to install packages like neeto-icons, neeto-molecules and neeto-ui which does not make sense for an end user since this package serves as a utility package to avoid unwanted boilerplate code. I also saw that the `antd` package was installed when I installed this package. We should decouple this repository from the unwanted dependencies so that it is straightforward.

cc// @josephmathew900, @AbhayVAshokan

### PR DESCRIPTION
**Pull request by soorajbhaskaran**
Fixes #46 

Updates:
- Removed all dependencies related to neeto-commons-frontend by handpicking only the required rules
- Added the config for doing a beta release at the workflows
- Created a beta release and tested it on a non-neeto project

#### Checklist

- [ ] I have made corresponding changes to the documentation
- [ ] I have verified the generated JSDoc comments
- [ ] I have updated the types definition of modified exports
- [ ] I have added jest tests for newly added changes
- [ ] I have verified the functionality in some of the neeto web-apps.

### CODE REVIEW

**Comment by soorajbhaskaran** [comment]
@josephmathew900 _a: Can you please check?

File: .eslintrc.js [hunk 1]
```diff
@@ -1,5 +1,27 @@
-const defaultConfigurations = require("@bigbinary/neeto-commons-frontend/configs/nanos/eslint/index.js");
-const { mergeDeepLeft } = require("ramda");
+const { mergeLeft, mergeDeepLeft } = require("ramda");
+
+const commonConfiguration = require("./configs/eslint");
+
+const nanosConfiguration = {
+  extends: [
+    "plugin:@bigbinary/neeto/nanos-recommended",
```
Comment (josephmathew900) [inline .eslintrc.js]: We can also remove the `eslint-plugin-neeto` package because it hasn't been open sourced yet.

**Review by josephmathew900 — APPROVED** [review]
@soorajbhaskaran _a Added a minor suggestion. Rest LGTM

**Review by josephmathew900 — APPROVED** [review]
LGTM

**Pull request merged**